cpp/benchmarks/io/orc/orc_writer_chunks.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmarks/common/generate_input.hpp>
+#include <benchmarks/fixture/benchmark_fixture.hpp>
+#include <benchmarks/io/cuio_common.hpp>
+#include <benchmarks/synchronization/synchronization.hpp>
+
+#include <cudf/column/column.hpp>
+#include <cudf/io/orc.hpp>
+#include <cudf/table/table.hpp>
+
+// to enable, run cmake with -DBUILD_BENCHMARKS=ON
+
+constexpr int64_t data_size = 512 << 20;
+
+namespace cudf_io = cudf::io;
+
+class OrcWrite : public cudf::benchmark {
+};
+class OrcWriteChunked : public cudf::benchmark {
+};
+
+void ORC_write(benchmark::State& state)
+{
+  cudf::size_type num_cols = state.range(0);
+
+  auto tbl              = create_random_table(cycle_dtypes({cudf::type_id::INT32,
+                                               cudf::type_id::FLOAT64,
+                                               cudf::type_id::STRING,
+                                               cudf::type_id::DECIMAL32,
+                                               cudf::type_id::INT8},
+                                              num_cols),
```

Comment:
You can cover all supported types:
```suggestion
  auto tbl              = create_random_table(cycle_dtypes(get_type_or_group({int32_t(type_group_id::INTEGRAL_SIGNED),
                                                   int32_t(type_group_id::FLOATING_POINT),
                                                   int32_t(type_group_id::FIXED_POINT),
                                                   int32_t(type_group_id::TIMESTAMP),
                                                   int32_t(cudf::type_id::STRING),
                                                   int32_t(cudf::type_id::STRUCT),
                                                   int32_t(cudf::type_id::LIST)},
                                              num_cols),
```